ode/src/KokkosODE_RungeKutta.hpp

```diff
@@ -0,0 +1,191 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include "Kokkos_Core.hpp"
+#include "KokkosODE_Types.hpp"
+
+#include "KokkosODE_RungeKutta_impl.hpp"
+
+#ifndef KOKKOSODE_RUNGEKUTTA_HPP
+#define KOKKOSODE_RUNGEKUTTA_HPP
+
+namespace KokkosODE {
+namespace Experimental {
+
+enum RK_type : int {
+  RKFE  = 0,
+  RKEH  = 1,
+  RKF12 = 2,
+  RKBS  = 3,
+  RK4   = 4,
+  RKF45 = 5,
+  RKCK  = 6,
+  RKDP  = 7
+};
+
+template <RK_type T>
+struct RungeKutta {};
```

Comment:
If you want to get rid of the copy-pasted specializations, you could do
```
template<RK_type T> typedef table_type void;
template<> typedef table_type<RKFE> KokkosODE::Impl::ButcherTableau<0, 0>;
template<> typedef table_type<RKEH> ...
```
and then one generic ``struct RungeKutta`` definition would work for all of them

Reply:
Never mind, you can't specialize a templated typedef, thanks C++. Structs work though:
```
template<RK_type T> struct TT {using type = void;};

template<> struct TT<RKFE> {using type = ButcherTableau<0, 0>;};
template<> struct TT<RKEH>  {using type = ButcherTableau<0, 1>;};
...
```
```
int o = TT<T>::order();
```